### ISSUE TITLE
Deprecate Unnecessary Modules

### ISSUE THREAD

**Issue by bhough** [styled-components/polished#175]
Mark modules for deprecation in 3.0 due to being rendered useless by improvements to CSS processors used by supported libraries:

- [x] selection
- [x] placeholder

**Comment by mxstbr** [comment]
Let's make these log a warning to the console saying something like "Hey, x is deprecated and will be removed in v3.0" and then for v3.0 delete them for real!

**Comment by bhough** [comment]
Sounds good @mxstbr that will make this issue dependent on #111, as it will introduce a way to send a deprecation message.

**Comment by nikgraf** [comment]
@bhough why deprecate `toColorString`? While not heavily used I think it's useful for people doing advanced color manipulation.

**Comment by bhough** [comment]
@nikgraf just my misunderstanding of what replacing it with `hslToColorString` and `rgbToColorString`. We can remove it from this list.

**Comment by bhough** [comment]
Addressed by #338 and merged into version-2.
